### PR TITLE
Amount interface for `InvoiceDetails`

### PR DESCRIPTION
**Pull request by andrei-21**

### CODE REVIEW

File: examples/3l-node/cli.rs [hunk 1]
```diff
@@ -138,7 +138,7 @@ fn setup_editor(history_path: &Path) -> Editor<CommandHinter, DefaultHistory> {
     hints.insert(CommandHint::new("nodeinfo", "nodeinfo"));
     hints.insert(CommandHint::new("lspfee", "lspfee"));
     hints.insert(CommandHint::new(
-        "calculatelspfee <amount in millisat>",
+        "calculatelspfee <amount in sat>",
```
Comment (danielgranhao) [inline examples/3l-node/cli.rs:141]:
```suggestion
        "calculatelspfee <amount in sats>",
```

File: examples/3l-node/cli.rs [hunk 2]
```diff
@@ -162,7 +162,7 @@ fn setup_editor(history_path: &Path) -> Editor<CommandHinter, DefaultHistory> {
     ));
     hints.insert(CommandHint::new("payinvoice <invoice>", "payinvoice "));
     hints.insert(CommandHint::new(
-        "payopeninvoice <invoice> <amount in msat>",
+        "payopeninvoice <invoice> <amount in sat>",
```
Comment (danielgranhao) [inline examples/3l-node/cli.rs:165]:
```suggestion
        "payopeninvoice <invoice> <amount in sats>",
```

File: examples/3l-node/cli.rs [hunk 3]
```diff
@@ -182,17 +182,17 @@ fn setup_editor(history_path: &Path) -> Editor<CommandHinter, DefaultHistory> {
 fn help() {
     println!("  nodeinfo");
     println!("  lspfee");
-    println!("  calculatelspfee <amount in millisat>");
+    println!("  calculatelspfee <amount in sat>");
```
Comment (danielgranhao) [inline examples/3l-node/cli.rs:185]:
```suggestion
    println!("  calculatelspfee <amount in sats>");
```

File: src/invoice_details.rs [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    pub description: String,
+    pub payment_hash: String,
+    pub payee_pub_key: String,
+    pub creation_timestamp: SystemTime,
+    pub expiry_interval: Duration,
+    pub expiry_timestamp: SystemTime,
+}
+
+impl InvoiceDetails {
+    pub(crate) fn from_our_invoice(invoice: Invoice, rate: &Option<ExchangeRate>) -> Self {
```
Comment (danielgranhao) [inline src/invoice_details.rs:19]: If you prefer "our", feel free to reject the suggestion.
```suggestion
    pub(crate) fn from_locally_issued_invoice(invoice: Invoice, rate: &Option<ExchangeRate>) -> Self {
```

File: src/lipalightninglib.udl [hunk 5]
```diff
@@ -253,7 +253,7 @@ interface LiquidityLimit {
 // Information embedded in an invoice
 dictionary InvoiceDetails {
     string invoice; // The BOLT11 invoice.
-    u64? amount_msat; // Payment amount in msats, if specified. If not available, invoice is an open-amount invoice
+    Amount? amount; // Payment amount, if specified. If not available, invoice is an open-amount invoice
```
Comment (danielgranhao) [inline src/lipalightninglib.udl:256]: Now that `InvoiceDetails` carries a fiat value, we need to make clear which exchange rate is used for that fiat value. At the moment, I'm worried about the following situation in which a discrepancy can arise:
1. Consumer creates an invoice and looks at the fiat value contained within `InvoiceDetails`
    * It is reasonable for the consumer to expect that this is the fiat value that will get persisted in the local records
2. Later, the consumer gets the list of `Payments` and the `InvoiceDetails` inside the related payment has a different fiat value.

Maybe we don't need to worry about this edge case

**Review by danielgranhao — APPROVED** [review]
Generally LGTM. Left some very minor suggestions. 

Maybe one

File: examples/3l-node/cli.rs (hunk 1 above)
Comment (gcomte) [reply examples/3l-node/cli.rs:141]: We've been very inconsistent with that. Myself I would prefer `SAT` in capital letters (for plural and singular). This is also how it's being displayed in the Lipa Wallet.

File: src/lipalightninglib.udl [hunk 6]
```diff
@@ -253,8 +253,12 @@ interface LiquidityLimit {
 // Information embedded in an invoice
 dictionary InvoiceDetails {
     string invoice; // The BOLT11 invoice.
-    u64? amount_msat; // Payment amount in msats, if specified. If not available, invoice is an open-amount invoice
-        // and the user should be promted for how much they want to pay.
+	// Payment amount, if specified. If not available, invoice is an open-amount invoice
+    // and the user should be promted for how much they want to pay.
+	// The fiat value is calculated base on the "natural" exchange rate:
+	//  - for a new invoice current exchange rate is used
+	//  - for old invoices historic values are used
```
Comment (danielgranhao) [inline src/lipalightninglib.udl:260]:
```suggestion
    // Payment amount, if specified. If not available, invoice is an open-amount invoice
    // and the user should be promted for how much they want to pay.
    // The fiat value is calculated base on the "natural" exchange rate:
    //  - for a new invoice current exchange rate is used
    //  - for old invoices historic values are used
```

**Review by danielgranhao — APPROVED** [review]

**Pull request merged**